core/src/main/java/org/pentaho/platform/scheduler2/quartz/EmbeddedQuartzSystemListener.java

```diff
@@ -163,6 +163,11 @@ public boolean startup( final IPentahoSession session ) {
     return result;
   }
 
+  protected void initializeScheduler()
+    throws org.pentaho.platform.api.scheduler2.SchedulerException {
+    PentahoSystem.get( IScheduler.class ).start();
```

Comment:
you need to be **_consistent_** on how you get the class that implements the class`IScheudler.class`

You are mixing two different approaches and if for some reason there are two instances of  `IScheudler.class` you will get different classes.

1. PentahoSystem.get( IScheduler.class ) - will return the implementing class with highest priority.
2. PentahoSystem.get( IScheduler.class, "IScheduler2", null ); - assumes their is only one class named "IScheduler2" and the return class will be undetermined if there are multiple instances.
   -  https://github.com/pentaho/pentaho-scheduler-plugin/blob/scheduler-plugin/core/src/main/java/org/pentaho/platform/scheduler2/quartz/EmbeddedQuartzSystemListener.java#L112

Reply:
I will stick with the existing convention of option 2